### DISCUSSION TITLE
unable to restore "ERROR:  invalid byte sequence for encoding "UTF8": 0xff"

### DISCUSSION THREAD

**Discussion by stdestro** [teslamate-org/teslamate#2555] [Q&A]
EDIT: turns out creating the backup in windows was the problem.
using: `iconv -f utf-16 -t utf-8 teslamate.bck > teslamate2.bck` did the trick

Hi, I'm migrating Teslamate to a different machine.
the old one is a docker istance on windows, the new one is docker on raspbian

I created the backup using

`docker-compose exec -T database pg_dump -U teslamate teslamate > /backuplocation/teslamate.bck`

tried to restore on the new machine using
`docker-compose exec -T database psql -U teslamate -d teslamate < teslamate.bck`

but I get this error message:
"ERROR:  invalid byte sequence for encoding "UTF8": 0xff"

the backup is huge, more than 1GB so i cannot open with notepad, and i tried to specify `--encoding` while creating the backup with no luck

Any chance someone had the same issue end could point me in the right direction?

**Comment by ekalp** [comment]
I am having the same issue moving from a Docker install on Docker Desktop (Windows) to Docker on Ubuntu. :(

**Comment by ekalp** [comment]
@stdestro I ended up fixing this manually with the following steps as the command you used didn't work for me. :(

1. On both instances I ran the following to ensure data collection was stopped: docker-compose stop teslamate
2. Installed pgAdmin on both Docker instances using the directions from here: https://hevodata.com/learn/docker-postgresql/#step3
3. Joined the pgAdmin container to the teslamate_default network using Portainer (if you don't do this step you can't connect to the postgresql instance, as it isn't exposed on the host network)
4. Connected pgAdmin to the teslamate postgresql install using the username/password from my docker-compose.yml on both Docker instances
5. Ran the second docker-compose statement from the restore directions here against my new Docker instance to prep the DB for the restore: https://docs.teslamate.org/docs/maintenance/backup_restore
6. Performed a backup of my teslamate db from my original instance using pgAdmin, Filename: TeslamateBackup.sql, Format: Custom, Compression ratio: 3, Encoding: UTF8, Role name: teslamate (You can change the compression ratio to shrink the file if needed.)
7. Performed a restore my teslamate db to my new instance using pgAdmin, Filename: TeslamateBackup.sql, Role name: teslamate
8. Ran the following on my new instance: docker-compose start teslamate
9. Verified that teslamate was now running as expected on newinstanceip:4000
10. Verified that grafana was now running as expected on newinstanceip:3000 (had to reset the password from the default since it is stored in the container storage, not the DB)
11. Verified that my data was intact using grafana
12. Disabled the teslamate container on the original docker instance from starting

**Comment by fabsenet** [comment]
So I got here, because I have the same issue! I want to move from windows to linux.

What I noticed, i was using powershell on windows to create the database dump and opening it with vscode revealed, it was utf16 and special characters like `ß` where already broken inside the file.

Solution was to use cmd ("kind of old dos prompt") running the same dump command. this created a utf8 file with special chars intact. I was able to import this file without any error and special characters where just fine!

**Reply by EnixCoda** [reply]
If anyone got here and has removed the database volume, like myself, that there is no chance to backup the database again.
You can convert the dump file `teslamate.bck` from `UTF-16LE` (default encoding of powershell) to `UTF-8`, via tools like `iconv`, to fix the issue.
